hibernate-core/src/test/java/org/hibernate/test/annotations/onetoone/hh11746/OneToOneCascadeMapsId.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.annotations.onetoone.hh11746;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Jonathan Ihm
+ */
+public class OneToOneCascadeMapsId extends BaseCoreFunctionalTestCase {
+    @Test
+    public void hhh11746Test() throws Exception {
+        // BaseCoreFunctionalTestCase automatically creates the SessionFactory and provides the Session.
+        Session s = openSession();
+        Transaction tx = s.beginTransaction();
+        insertTestData(s);
+        tx.commit();
+
+        s.clear();
+        tx = s.beginTransaction();
+        Person person = s.find(Person.class, new PersonId("a1", "s1"));
+        Address adr = person.getAddress();
+        assertEquals(person.getId(), adr.getId());
+
+        person = s.find(Person.class, new PersonId("a2", "s2"));
+        adr = person.getAddress();
+        assertEquals(person.getId(), adr.getId());
```

Comment:
The two assertions here should probably read:
<pre>assertEquals( person.getId().getId(), adr.getId() );</pre>